### PR TITLE
[TDD] Add edgesrc/sink and query custom connection tests

### PR DESCRIPTION
**Pull request by gichan-jang**
Add edgesrc/sink and tensor_qeury custom connection tests. Enable the tests later.

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: tests/nnstreamer_edge/edge/unittest_edge.cc [hunk 1]
```diff
@@ -358,6 +359,169 @@ TEST (edgeSinkSrc, runNormalAitt)
...
+  GstElement *gstpipe;
+
+  /* Create a nnstreamer pipeline */
+  pipeline = g_strdup_printf (
+      "gst-launch-1.0 videotestsrc ! videoconvert ! videoscale ! "
+      "video/x-raw,width=320,height=240,format=RGB,framerate=10/1 ! "
+      "tensor_converter ! edgesink connect-type=CUSTOM custom-lib=%s name=sinkx port=0",
+      CUSTOM_LIB_PATH);
+  gstpipe = gst_parse_launch (pipeline, NULL);
+  EXPECT_NE (gstpipe, nullptr);
```
Comment (songgot) [inline tests/nnstreamer_edge/edge/unittest_edge.cc:381]: Well written TC. It seems necessary to initialize ```gstpipe``` to nullptr for nullptr comparison.

File: tests/nnstreamer_edge/edge/unittest_edge.cc [hunk 2]
```diff
@@ -358,6 +359,169 @@ TEST (edgeSinkSrc, runNormalAitt)
...
 }
 #endif
+
+/**
+ * @brief Test for edgesink custom connection.
+ */
+TEST (edgeCustom, sinkNormal)
+{
+  /** @todo TDD: Enable this test later. */
+  GTEST_SKIP ();
```
Comment (songgot) [inline tests/nnstreamer_edge/edge/unittest_edge.cc:369]: You wrote a TDD-based TEST, that's great. :1st_place_medal:

File: tests/nnstreamer_edge/edge/unittest_edge.cc (hunk 1 above)
Comment (gichan-jang) [reply tests/nnstreamer_edge/edge/unittest_edge.cc:381]: Updated. Thanks!

File: tests/nnstreamer_edge/query/unittest_query.cc [hunk 3]
```diff
@@ -266,6 +208,244 @@ TEST (tensorQuery, clientAlone_n)
   g_free (pipeline);
 }
 
+
+/**
+ * @brief Test for tensor_query_server custom coonnection
```
Comment (niley7464) [inline tests/nnstreamer_edge/query/unittest_query.cc:213]:
```suggestion
 * @brief Test for tensor_query_server custom connection
```

Hmm,, Why didn't the typo action detect it 🤔?

File: tests/nnstreamer_edge/query/unittest_query.cc [hunk 4]
```diff
@@ -266,6 +208,244 @@ TEST (tensorQuery, clientAlone_n)
...
+
+  EXPECT_EQ (setPipelineStateSync (gstpipe, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (100000);
+
+  gst_object_unref (gstpipe);
+  g_free (pipeline);
+}
+
+/**
+ * @brief Test for tensor_query_server custom coonnection with invalid property.
```
Comment (niley7464) [inline tests/nnstreamer_edge/query/unittest_query.cc:247]: Ditto

**Review by niley7464 — APPROVED** [review]
This would be a good example of using TDD techniques 👍 
LGTM except for minor comment

**Review by jaeyun-jung — APPROVED** [review]

File: tests/nnstreamer_edge/query/unittest_query.cc (hunk 3 above)
Comment (gichan-jang) [reply tests/nnstreamer_edge/query/unittest_query.cc:213]: It's bug..😥
I tested it on my personal github action, it's not detected.

Changes with typos: coonnection Finallizes
```
diff --git a/tests/nnstreamer_edge/query/unittest_query.cc b/tests/nnstreamer_edge/query/unittest_query.cc
index 08641068..eaa9d4b8 100644
--- a/tests/nnstreamer_edge/query/unittest_query.cc
+++ b/tests/nnstreamer_edge/query/unittest_query.cc
@@ -210,7 +210,7 @@ TEST (tensorQuery, clientAlone_n)
 
 
 /**
- * @brief Test for tensor_query_server custom coonnection
+ * @brief Test for tensor_query_server custom coonnection Finallizes
  */
 TEST (tensorQuery, serverCustomNormal)
 {
```
Detected result:
```
./typos
jq: jq-1.6
$ ./typos . --config ./.github/workflows/_typos.toml
Warning: "Finallizes" should be "Finalizes".
error: `Finallizes` should be `Finalizes`
  --> ./tests/nnstreamer_edge/query/unittest_query.cc:213:59
    |
213 |  * @brief Test for tensor_query_server custom coonnection Finallizes
    |                                                           ^^^^^^^^^^
    |
Error: Process completed with exit code 2.
```

I request to fix it. https://github.com/crate-ci/typos/issues/1090

File: tests/nnstreamer_edge/query/unittest_query.cc (hunk 4 above)
Comment (gichan-jang) [reply tests/nnstreamer_edge/query/unittest_query.cc]: Updated.

File: tests/nnstreamer_edge/edge/unittest_edge.cc (hunk 1 above)
Comment (songgot) [reply tests/nnstreamer_edge/edge/unittest_edge.cc:381]: Thank you.

**Review by songgot — APPROVED** [review]
LGTM :100:

**Pull request merged**